xcape/core.py

```diff
@@ -153,55 +153,46 @@ def calc_cape(p, t, td, ps, ts, tds, *args, **kwargs):
         return _calc_cape_numpy(p, t, td, ps, ts, tds, *args, **kwargs)
     
 
-def _calc_cape_gufunc(p, t, td, ps, ts, tds, *args, **kwargs):
+def _calc_cape_gufunc(*args, **kwargs):
     
     if (kwargs['vertical_lev']=='sigma'):
         signature = "(i),(i),(i),(),(),()->(),()"
         output_dtypes = ('f4','f4')
-        if kwargs['source']=='most-unstable':
-            signature += ",(),()"
-            output_dtypes = output_dtypes + ('i4','f4')
-        return da.apply_gufunc(_calc_cape_numpy, signature,
-                               p, t, td, ps, ts, tds, 
-                               output_dtypes=output_dtypes,#('f4','f4'),
-                               axis=-1,
-                               vectorize=False,
-                               **kwargs)
-    
     elif (kwargs['vertical_lev']=='pressure'):
-        pres_lev_pos = args[0]
         signature = "(i),(i),(i),(),(),(),()->(),()"
         output_dtypes = ('f4','f4')
-        if kwargs['source']=='most-unstable':
-            signature += ",(),()"
-            output_dtypes = output_dtypes + ('i4','f4')
-        return da.apply_gufunc(_calc_cape_numpy, signature,
-                               p, t, td, ps, ts, tds, pres_lev_pos,
-                               output_dtypes=output_dtypes,#('f4','f4'),
+        
+    if kwargs['source']=='most-unstable':
+        signature += ",(),()"
+        output_dtypes = output_dtypes + ('i4','f4')
+
+    return da.apply_gufunc(_calc_cape_numpy, signature,
+                               *args, 
+                               output_dtypes=output_dtypes,
                                axis=-1,
                                vectorize=False,
                                **kwargs)
-
+    
 # the numpy version of the algorithm
-def _calc_cape_numpy(p, t, td, ps, ts, tds, *args, 
+def _calc_cape_numpy(*args, 
                      source='surface', ml_depth=500.,
                      adiabat='pseudo-liquid',pinc=500., method='fortran',
                      vertical_lev='sigma'):
 
-
-    original_shape = p.shape
-    original_surface_shape = ps.shape
+    p, t, td, ps, ts, tds, *pres_lev_pos_in = args
+    
+    original_shape = p.shape #shape of 3D variable, i.e. p
+    original_surface_shape = ps.shape #shape of surface variable, i.e. ps
 
     # after this, all arrays are 2d shape (nlevs, npoints)
     p_2d, t_2d, td_2d = _reshape_inputs(p, t, td)
     
-    if len(args)==0:
-        pres_lev_pos=1
-        p_s1d, t_s1d, td_s1d = _reshape_surface_inputs(ps, ts, tds)
-    elif len(args)==1:
-        p_s1d, t_s1d, td_s1d, pres_lev_pos = _reshape_surface_inputs(ps, ts, tds, 
-                                                                     args[0]+1) # to fortran convention
-
+    p_s1d, t_s1d, td_s1d, *pres_lev_pos = _reshape_surface_inputs(ps, ts, tds, *pres_lev_pos_in) 
+    if len(args)==7:
+        pres_lev_pos = pres_lev_pos[0]+1 # to fortran convention
+    elif len(args)==6:
+        pres_lev_pos = 1
+        
```

Comment:
so here is the clunky part. I guess because it is an optional argument, pres_lev_pos_in gets evaluated separately, so it comes out as a list of one element and I need to puyt the [0] 
`pres_lev_pos = pres_lev_pos[0]+1`

because of this, I cannot just have this operation without the if, and perhaps define a pres_ev_pos = 0 at the beginning, because it would fail. so I still need this if loop. Which is fine, but maybe there is a way out.